test/resolvers/project-version.js

```diff
@@ -101,53 +120,99 @@ describe('ProjectVersion resolver', () => {
-        .then(version => {
-          assert.equal(version.raCompulsory, true);
-        });
-    });
-
     describe('retrospective assessment', () => {
 
+      it('sets ra flag to true if training licence', () => {
+        const opts = {
+          action: 'patch',
+          id: draftVersionId,
+          data: {
+            patch: jsondiff.diff({}, { 'training-licence': true })
+          }
+        };
+        return Promise.resolve()
+          .then(() => this.projectVersion(opts))
+          .then(() => this.models.ProjectVersion.query().findById(draftVersionId))
+          .then(version => {
+            assert.equal(version.raCompulsory, true);
+          });
+      });
+
       it('flags raCompulsory if severe protocol is added', () => {
         const opts = {
           action: 'patch',
-          id: versionId,
+          id: draftVersionId,
           data: {
             patch: jsondiff.diff({}, { protocols: [{ severity: 'mild' }, { severity: 'severe' }] })
           }
         };
         return Promise.resolve()
           .then(() => this.projectVersion(opts))
-          .then(() => this.models.ProjectVersion.query().findById(versionId))
+          .then(() => this.models.ProjectVersion.query().findById(draftVersionId))
           .then(version => {
             assert.equal(version.raCompulsory, true);
           });
       });
 
       it('does not flag raCompulsory if severe protocol is deleted', () => {
+        const opts = {
+          action: 'patch',
+          id: draftVersionId,
+          data: {
+            patch: jsondiff.diff({}, { protocols: [{ severity: 'mild' }, { severity: 'severe', deleted: true }] })
+          }
+        };
+        return Promise.resolve()
+          .then(() => this.projectVersion(opts))
+          .then(() => this.models.ProjectVersion.query().findById(draftVersionId))
+          .then(version => {
+            assert.equal(version.raCompulsory, false);
+          });
+      });
+
+      it('does not remove raCompulsory flag if project is not a draft', () => {
         const opts = {
           action: 'patch',
           id: versionId,
           data: {
             patch: jsondiff.diff({}, { protocols: [{ severity: 'mild' }, { severity: 'severe', deleted: true }] })
           }
         };
+        const patch = {
+          data: {
+            protocols: [{ severity: 'mild' }, { severity: 'severe' }]
+          },
+          raCompulsory: true
+        };
         return Promise.resolve()
+          .then(() => this.models.ProjectVersion.query().patchAndFetchById(versionId, patch))
           .then(() => this.projectVersion(opts))
           .then(() => this.models.ProjectVersion.query().findById(versionId))
           .then(version => {
-            assert.equal(version.raCompulsory, false);
+            assert.equal(version.raCompulsory, true);
+          });
+      });
+
+      it('adds raCompulsory flag if project is not a draft', () => {
```

Comment:
Not sure what's going on here because the project record inserted above with id `versionId` _is_ a draft.

Either the test description or the data needs to be updated I think.

Reply:
Nope. I'm misreading it. The _version_ is a draft, but the project is active.

FML.